### PR TITLE
[4346][ADD] procurement_mto_analytic

### PR DESCRIPTION
**Pull request by yostashiro**
[4346](https://www.quartile.co/web#id=4346&menu_id=505&cids=3&action=1457&model=project.task&view_type=form)

### CODE REVIEW

File: procurement_mto_analytic/models/stock_rule.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+from odoo import models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    def _find_matching_analytic_distribution_record(self, analytic_distribution):
+        # Prepare the JSONB structure as a string for the SQL query
+        analytic_distribution_str = json.dumps(analytic_distribution)
```
Comment (AungKoKoLin1997) [inline procurement_mto_analytic/models/stock_rule.py]: @yostashiro I checked the behavior of how analytic_distribution sequence works. We don't need to worry about the sequence. It is always sequence by projects and multiple analytic_distributions for one project are also sequence by themselves even we don't fill in order.

File: procurement_mto_analytic/models/stock_rule.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                lambda x: x.analytic_distribution == values.get("analytic_distribution")
+            )
+        )
+        if matching_lines:
+            domain += (("order_line", "in", tuple(matching_lines.ids)),)
+        else:
+            domain += (
+                ("order_line", "=", False),
+            )  # This might need adjustment based on your intent.
+        return domain
```
Comment (yostashiro) [inline procurement_mto_analytic/models/stock_rule.py]: Reviewing the design, I now doubt that this should be the way to go, as there can be cases where one purchase order holds mutplie analytic distributions. We should not be extending this method.

We should instead create a PR in odoo/odoo to split https://github.com/OCA/OCB/blob/1e9b8db/addons/purchase_stock/models/stock_rule.py#L125 into a method, and extend that method in this module to add analytic distribution as a key.

Comment (yostashiro) [reply procurement_mto_analytic/models/stock_rule.py]: To clarify, this design is fine for the client (as there should be one analytic distribution per order), therefore we will keep it as is in this repo for now.  We will probably split this part into another (non-OCA) custom module once the OCA PR gets merged.

**Comment by yostashiro** [comment]
@AungKoKoLin1997 Can you revert your last commit for now? See https://github.com/qrtl/axls-oca/pull/119#discussion_r1563834390. This update will bring confusion to the client if we don't do it together with a solution to add analytic distribution as a key to group purchase orders.

**Comment by AungKoKoLin1997** [comment]
> Can you revert your last commit for now?

Done! Sorry, I updated it by seeing comments from OCA. I missed the new comment from this repo.

**Pull request merged**